torchmdnet/models/utils.py

```diff
@@ -434,6 +434,49 @@ def forward(self, distances: Tensor) -> Tensor:
             return cutoffs
 
 
+class MLP(nn.Module):
+    """A simple multi-layer perceptron with a given number of layers and hidden channels.
+
+    Args:
+        in_channels (int): Number of input features.
+        out_channels (int): Number of output features.
+        hidden_channels (int): Number of hidden features.
+        activation (str): Activation function to use.
+        num_layers (int, optional): Number of layers. Defaults to 0.
+        dtype (torch.dtype, optional): Data type to use. Defaults to torch.float32.
+    """
+
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        hidden_channels,
+        activation,
+        num_layers=0,
```

Comment:
Default before was 2, no?

Reply:
Actually this might be a misconception on my part regarding naming.
num_layers=0 means:
act(linear): in_channels -> hidden_channels
linear: hidden_channels -> out_channels

num_layers=1 would be:
act(linear): in_channels -> hidden_channels
act(linear): hidden_channels->hidden_channels
linear: hidden_channels -> out_channels

It makes sense to me because otherwise num_layers<2 would not be possible. But maybe my naming is not the standard?

Reply:
I like this convention:

num_layers=1 would be:
act(linear): in_channels -> hidden_channels
act(linear): hidden_channels->hidden_channels
linear: hidden_channels -> out_channels

But I would call them hidden layers, instead of layers.

Reply:
The first one is the input layer, the last one the output layer.

Reply:
Changed the name to num_hidden_layers and clarified in the docstring.